editor/src/components/canvas/dom-walker.spec.browser2.tsx

```diff
@@ -406,6 +409,166 @@ describe('Capturing closest offset parent', () => {
+describe('Observing runtime changes', () => {
+  const changingProjectCode = `
+    import * as React from 'react'
+    import { Scene, Storyboard } from 'utopia-api'
+
+    const App = (props) => {
+      const [showDiv, setShowDiv] = React.useState(false)
+
+      return (
+        <div
+          style={{ width: '100%', height: '100%' }}
+          data-uid='app-root'
+        >
+          <div
+            style={{
+              backgroundColor: 'lightblue',
+              position: 'absolute',
+              left: 49,
+              top: 59,
+              width: 200,
+              height: 44,
+              textAlign: 'center',
+            }}
+            onMouseUp={() => {
+              setTimeout(() => setShowDiv(true), 10) // setTimeout so that this happens after we handle mouse events
+            }}
+            data-uid='button'
+            data-testid='click-me'
+          >
+            Click me
+          </div>
+          {
+            // @utopia/uid=conditional
+            showDiv ? (
+            <div
+              style={{
+                backgroundColor: 'lightblue',
+                position: 'absolute',
+                left: 49,
+                top: 150,
+                width: 200,
+                height: 44,
+                textAlign: 'center',
+              }}
+              data-uid='target-div'
+              data-testid='target-div'
+            >
+              Hello!
+            </div>
+          ) : null}
+        </div>
+      )
+    }
+
+    export var storyboard = (
+      <Storyboard data-uid='sb'>
+        <Scene
+          style={{
+            backgroundColor: '#aaaaaa33',
+            position: 'absolute',
+            left: 532,
+            top: 66,
+            width: 296,
+            height: 457,
+          }}
+          data-uid='sc'
+        >
+          <App data-uid='app' />
+        </Scene>
+      </Storyboard>
+    )
+  `
+
+  it('Updates the metadata in live mode', async () => {
+    const renderResult = await renderTestEditorWithCode(
+      changingProjectCode,
+      'await-first-dom-report',
+    )
+    await renderResult.dispatch([switchEditorMode(EditorModes.liveMode())], true)
+    await renderResult.getDispatchFollowUpActionsFinished()
+
+    const targetPath = EP.fromString('sb/sc/app:app-root/conditional/target-div')
+
+    // Check no metadata for the target at the start
+    const metadataBefore = MetadataUtils.findElementByElementPath(
+      renderResult.getEditorState().editor.jsxMetadata,
+      targetPath,
+    )
+    expect(metadataBefore).toBeNull()
+
+    const recordedActionsBefore = [...renderResult.getRecordedActions()]
+
+    const buttonToClick = renderResult.renderedDOM.getByTestId('click-me')
+    const buttonToClickBounds = buttonToClick.getBoundingClientRect()
+    const clickPoint = {
+      x: buttonToClickBounds.width / 2,
+      y: buttonToClickBounds.height / 2,
+    }
+
+    await mouseClickAtPoint(buttonToClick, clickPoint)
+    await wait(20)
```

Comment:
can this be await(0) or awaiting a requestAnimationFrame? I have a feeling we are doing us a disservice by putting non-zero awaits into the test code – maybe it's superstition on my part, but it certainly something that can become flaky in the future

Reply:
tbh reducing the setTimeout to zero hadn't occurred to me. I've tried that and it works perfectly (including reproducing the issue that this PR fixes), so I've made that change 👍